tests/test_download.py

```diff
@@ -82,6 +91,17 @@ def test_resiliparse_extract_text(self):
 
         assert result == expected
 
+    def test_trafilatura_extract_text(self, html_string):
+        algorithm = TrafilaturaExtractor()
+        stop_words = get_stop_list_dict()
+        result = algorithm.extract_text(html_string, stop_words["ENGLISH"])
+
+        expected = [
+            "Let's keep this paragraph: either came does last new took taken making became from.",
+        ]
```

Comment:
Trafilatura has a really bad bug where it is returning the string twice. I can double check all of my logic, but in the case that this is a Trafilatura-specific issue, I am debating whether I should add our own exact deduplication code into this.

(Trafilatura actually has their own `deduplicate` parameter [here](https://github.com/adbar/trafilatura/blob/b0107793454f7c6334d87f1228e0ff24d19c02b7/trafilatura/core.py#L365), but it did not fix this use case.)

Possible related issue: https://github.com/adbar/trafilatura/issues/634

Reply:
I have not scoped to see what conditions cause this.

Reply:
Another issue: https://github.com/adbar/trafilatura/issues/768

Reply:
Solved by adding support for these extraction and deduplication parameters: https://trafilatura.readthedocs.io/en/latest/settings.html.